packages/wallets/src/components/WalletsCarouselContent/WalletsCarouselContent.tsx

```diff
@@ -11,60 +11,76 @@ type TProps = {
+                // exception from general rule of Embla being single source of truth
+                // as this is needed to set the initial active wallet
+                setSelectedEmblaIndex(activeWalletIndex);
+                walletsCarouselEmblaApi?.scrollTo(activeWalletIndex, true);
+            }
         }
-    }, [walletsCarouselEmblaApi]);
 
-    const { data: activeWallet } = useActiveWalletAccount();
-    const activeWalletIndex = useMemo(
-        () =>
-            walletAccountsList?.findIndex(wallet => wallet.loginid === activeWallet?.loginid) ??
-            walletsCarouselEmblaApi?.selectedScrollSnap() ??
-            0,
-        [activeWallet?.loginid, walletAccountsList, walletsCarouselEmblaApi]
-    );
-
-    const [progressBarActiveIndex, setProgressBarActiveIndex] = useState(activeWalletIndex + 1);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [activeWallet, isActiveWalletLoading, walletsCarouselEmblaApi]);
 
+    // bind to embla events
     useEffect(() => {
-        if (isCarouselInitialized) {
-            walletsCarouselEmblaApi?.scrollTo(activeWalletIndex);
-            setProgressBarActiveIndex(activeWalletIndex + 1);
+        if (!walletsCarouselEmblaApi) {
+            return;
         }
-    }, [activeWalletIndex, isCarouselInitialized, walletsCarouselEmblaApi]);
 
-    useEffect(() => {
-        walletsCarouselEmblaApi?.on('settle', () => {
+        walletsCarouselEmblaApi.on('select', () => {
             const scrollSnapIndex = walletsCarouselEmblaApi?.selectedScrollSnap();
-            const loginid = walletAccountsList?.[scrollSnapIndex]?.loginid;
-            if (activeWallet?.loginid !== loginid) {
-                switchAccount(loginid || '');
-            }
-            onWalletSettled?.(true);
+            setSelectedEmblaIndex(scrollSnapIndex);
         });
 
-        walletsCarouselEmblaApi?.on('select', () => {
-            const scrollSnapIndex = walletsCarouselEmblaApi?.selectedScrollSnap();
-            setProgressBarActiveIndex(scrollSnapIndex + 1);
-            onWalletSettled?.(false);
+        // on settle, this is only for tutorial / onboarding plugin in some other components,
+        walletsCarouselEmblaApi.on('settle', () => {
+            onWalletSettled && onWalletSettled(true);
         });
-    }, [walletsCarouselEmblaApi, switchAccount, walletAccountsList, activeWallet?.loginid, onWalletSettled]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [walletsCarouselEmblaApi]);
 
-    const amountOfSteps = useMemo(() => walletAccountsList?.map(wallet => wallet.loginid), [walletAccountsList]);
+    // load active wallet whenever its scrolled
+    useEffect(() => {
+        const selectedLoginId = walletAccountsList?.[selectedEmblaIndex]?.loginid;
+        if (selectedLoginId) {
+            switchAccount(selectedLoginId);
+        }
+    }, [selectedEmblaIndex, switchAccount, walletAccountsList]);
```

Comment:
hey @wojciech-deriv 
i can see that we extracted 
```
const selectedLoginId = walletAccountsList?.[selectedEmblaIndex]?.loginid;
        if (selectedLoginId) {
            switchAccount(selectedLoginId);
        }
```
into useEffect.
just a question, is there any reason why we cannot run this logic in walletsCarouselEmblaApi.on('select') handler?

Reply:
I think it should work the same way. 
In my mind, that was done only for clarity/better separation/additional layer of safety (e.g. prevent duplicated calls if something goes wrong).
Though it should work the same way.

Reply:
basically - no solid reason for that specific aprroach, just reflection of how I modelled stuff in my brain :D

Reply:
got it @wojciech-deriv . the reason why i raised this query, because when i read new version of react docs, they strongly recommend to avoid useEffects as much as it possible, in that case we can replace the useEffect with event handler itself this is a [explanation](https://github.com/binary-com/deriv-app/pull/12412). but it just a suggestion, since we are getting benefits in  clarity/better separation/additional layer of safety of course we can proceed with your solution